Removes new line at the end of each cell

neffmallon opened the pull request:
resolves #3.

Ending a python file with a new line is important for version control software, but in ipython notebook cells, each cell is stored on a single line in the JSON. This commit removes the trailing new line character from cells without modifying the underlying black package.

Feel free to disregard or ask for more input from other users. :smiley:

neffmallon committed:
Removes new line at the end of each cell

Ending a python file with a new line is important for version control software, but in ipython notebook cells, each cell is stored on a single line in the JSON. This commit removes the trailing new line character from cells without modifying the underlying black package.

csurfer commented on blackcellmagic.py:
```diff
@@ -38,7 +38,7 @@ def black(self, line, cell):
         if cell:
             self.shell.set_next_input(
-                format_str(src_contents=cell, line_length=line_length),
+                format_str(src_contents=cell, line_length=line_length)[:-1],
```
This is a perfectly valid ask. Lets just be cautious here and check that the last char is indeed `\n` before removing it so that the package doesn't break even if black decides to remove the empty line at the file sometime in future.

neffmallon replied:
Good idea. New commit is tested.

neffmallon committed:
Check that end line is new line

If only a series of "\n" are passed, blackened is an empty string which has no last index, so we first must check that that blackened isn't an empty string.

csurfer commented on blackcellmagic.py:
```diff
@@ -37,8 +37,12 @@ def black(self, line, cell):
         args = magic_arguments.parse_argstring(self.black, line)
         line_length = args.line_length
         if cell:
+            blackened = format_str(src_contents=cell, line_length=line_length)
```
You are doing the work twice in line 40 and in line 45. And condition chaining in line 41,42 need not be nested. You can just do this and achieve the same.

```python
formatted = format_str(src_contents=cell, line_length=line_length)
if formatted and (formatted[-1] == "\n"):
    formatted = formatted[:-1]
self.shell.set_next_input(formatted, replace=True)
```

neffmallon committed:
Fixed endline check

neffmallon commented:
Yup, you were right. I missed doing the work twice. And I had the conditional the other way around first, so of course it was raising errors, which is why I nested. But yes, you are right again there.

Thanks!

bsamseth commented:
@csurfer This behavior is still seen, even though this fix has been merged. This is the case for the `0.0.1` version obtained through `pip install blackcellmagic`. Looking at [Pypi.org history](https://pypi.org/project/blackcellmagic/#history) it's clear that the last upload was the 25th of April 2018. Would you consider uploading an updated version?
